plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,508 @@
+                mso_template, mso, report_policy_route_map.get("template"), report_policy_route_map.get("name")
+            )
+
+        if static_report_route_map and not static_report_route_map_uuid:
+            static_report_route_map_uuid = get_route_map_policy_uuid_from_name(
+                mso_template, mso, static_report_route_map.get("template"), static_report_route_map.get("name")
+            )
+
+        mso_values = dict(
+            name=name,
+            description=description,
+            enableV3Asm=version3_asm,
+            enableFastLeaveControl=fast_leave,
+            enableReportLinkLocalGroups=report_link_local_groups,
+            igmpQuerierVersion=igmp_version,
+            groupTimeout=group_timeout,
+            queryInterval=query_interval,
+            queryResponseInterval=query_response_interval,
+            lastMemberCount=last_member_count,
+            lastMemberResponseInterval=last_member_response_time,
+            startQueryCount=startup_query_count,
+            startQueryInterval=startup_query_interval,
+            querierTimeout=querier_timeout,
+            robustnessFactor=robustness_variable,
+            stateLimitRouteMapRef=state_limit_route_map_uuid,
+            reportPolicyRouteMapRef=report_policy_route_map_uuid,
+            staticReportRouteMapRef=static_report_route_map_uuid,
+            maximumMulticastEntries=maximum_multicast_entries,
+            reservedMulticastEntries=reserved_multicast_entries,
+        )
+
+        if mso.existing:
+            append_update_ops_data(ops, match.details, igmp_interface_policy_attrs_path, mso_values)
+            mso.sanitize(match.details, collate=True)
+        else:
+            mso.sanitize(mso_values)
+            ops.append(dict(op="add", path="/tenantPolicyTemplate/template/igmpInterfacePolicies/-", value=mso.sent))
+
+    elif state == "absent":
+        if mso.existing:
+            ops.append(dict(op="remove", path=igmp_interface_policy_attrs_path))
+
+    if not module.check_mode and ops:
+        response_object = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        existing_mld_interface_policies = response_object.get("tenantPolicyTemplate", {}).get("template", {}).get("igmpInterfacePolicies", [])
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description, existing_mld_interface_policies, [KVPair("uuid", uuid) if uuid else KVPair("name", name)]
+        )
+        if match:
+            mso.existing = match.details  # When the state is present
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        mso.existing = mso.proposed if state == "present" else {}
+
+    mso.exit_json()
+
+
+def get_route_map_policy_uuid_from_name(mso_template, mso, template, name):
+    route_map_policy_for_multicast = MSOTemplate(mso, "tenant", template)
```

Comment:
Should we cache the `MSOTemplate` objects with the same template name, just so the modules doesn't need to get the template multiple times?

Reply:
yes, changed accordingly